src/Core/System/Language/LanguageLoader.php

```diff
@@ -26,6 +26,28 @@ public function loadLanguages(): array
             ->execute()
             ->fetchAll();
 
-        return FetchModeHelper::groupUnique($data);
+        $languages = FetchModeHelper::groupUnique($data);
+
+        return $this->resolveLocaleCodes($languages);
+    }
+
+    private function resolveLocaleCodes(array $languages): array
+    {
+        foreach ($languages as $languageId => $language) {
+            if ($language['code'] !== null) {
+                continue;
+            }
+
+            $currentLanguageId = $languageId;
+            while ($language['code'] === null) {
+                $currentLanguageId = $languages[$currentLanguageId]['parentId'];
+
+                $language['code'] = $languages[$currentLanguageId]['code'];
+            }
+
+            $languages[$languageId]['code'] = $language['code'];
```

Comment:
I'm not sure what php does here if you manipulate the array you currently iterate over :thinking: 

You could pass `$language` by reference into the loop, that way you won't need this line anymore

```php
foreach ($languages as $languageId => &$language) {
```

What do you think?

Reply:
I missed, that a child language must not have another child. Therefore, a recursive resolver is not necessary. I just moved that to the sql statement.